packages/base/matrix-room.gts

```diff
@@ -0,0 +1,415 @@
+        @name={{@model.author.displayName}}
+        @datetime={{formatRFC3339 this.timestamp}}
+        style={{cssVar
+          boxel-message-avatar-size=messageStyle.boxelMessageAvatarSize
+          boxel-message-meta-height=messageStyle.boxelMessageMetaHeight
+          boxel-message-gap=messageStyle.boxelMessageGap
+          boxel-message-margin-left=messageStyle.boxelMessageMarginLeft
+        }}
+      >
+        {{! template-lint-disable no-triple-curlies }}
+        {{{@model.formattedMessage}}}
+
+        {{#if @model.attachedCard}}
+          <this.cardComponent />
+        {{/if}}
+      </BoxelMessage>
+    </template>
+
+    get timestamp() {
+      if (!this.args.model.created) {
+        throw new Error(`message created time is undefined`);
+      }
+      return this.args.model.created.getTime();
+    }
+
+    get cardComponent() {
+      if (!this.args.model.attachedCard) {
+        return;
+      }
+      return this.args.model.attachedCard.constructor.getComponent(
+        this.args.model.attachedCard,
+        'isolated'
+      );
+    }
+  };
+  // The edit template is meant to be read-only, this field card is not mutable
+  static edit = class Edit extends JSONView {};
+}
+
+export class MatrixRoomCard extends Card {
+  // the only writeable field for this card should be the "events" field.
+  // All other fields should derive from the "events" field.
+  @field events = containsMany(MatrixEventCard);
+
+  @field roomId = contains(StringCard, {
+    computeVia: function (this: MatrixRoomCard) {
+      return this.events.length > 0 ? this.events[0].room_id : undefined;
+    },
+  });
+
+  @field name = contains(StringCard, {
+    computeVia: function (this: MatrixRoomCard) {
+      let events = this.events
+        .filter((e) => e.type === 'm.room.name')
+        .sort((a, b) => a.origin_server_ts - b.origin_server_ts) as
+        | RoomNameEvent[];
+      if (events.length > 0) {
+        return events.pop()!.content.name;
+      }
+      return; // this should never happen
+    },
+  });
+
+  @field creator = contains(RoomMemberCard, {
+    computeVia: function (this: MatrixRoomCard) {
+      let event = this.events.find((e) => e.type === 'm.room.create') as
+        | RoomCreateEvent
+        | undefined;
+      if (event) {
+        return upsertRoomMember(event.sender);
+      }
+      return; // this should never happen
+    },
+  });
+
+  @field created = contains(DateTimeCard, {
+    computeVia: function (this: MatrixRoomCard) {
+      let event = this.events.find((e) => e.type === 'm.room.create') as
+        | RoomCreateEvent
+        | undefined;
+      if (event) {
+        let timestamp = event.origin_server_ts;
+        return new Date(timestamp);
+      }
+      return; // this should never happen
+    },
+  });
+
+  @field messages = containsMany(MessageCard, {
+    isUsed: true, // TODO we should not have to set this--need to research this issue
+    computeVia: async function (this: MatrixRoomCard) {
```

Comment:
there is a runtime bug where for some reason the API thinks the `messages` field is not used in the template and as a result the computed is never run. I added this simple band-aid for now. But I'll follow up after this PR to find and fix this problem